dashboard/app/models/user.rb

```diff
@@ -836,7 +836,7 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level, new_result, submitted)
+  def track_level_progress_async(script_level, new_result, submitted, level_source_id)
```

Comment:
maybe we should just change this method to use keyword arguments (and requiring them for the required ones) since it has so many now (I'm adding another one for pair programming)